core/src/main/java/hudson/ExtensionList.java

```diff
@@ -246,8 +246,8 @@ private synchronized boolean removeSync(Object o) {
         return removed;
     }
 
-    private <T> boolean removeComponent(Collection<ExtensionComponent<T>> collection, Object t) {
-        for (ExtensionComponent<T> c : collection) {
+    private <X> boolean removeComponent(Collection<ExtensionComponent<X>> collection, Object t) {
+        for (ExtensionComponent<X> c : collection) {
             if (c.getInstance().equals(t)) {
```

Comment:
```suggestion
    private <X> boolean removeComponent(Collection<ExtensionComponent<X>> collection, Object x) {
        for (ExtensionComponent<X> c : collection) {
            if (c.getInstance().equals(x)) {
```

I would expect this to be called `t` because it's "a `T`" (or otherwise `T` related), so should be changed accordingly (or to something else entirely)

Reply:
Better yet, this doesn't need to be a generic method in the first place, because the only values being passed in by callers are already of type `T`. So we can keep using `T` and just remove the type parameter from the method.